src/answers-headless.ts

```diff
@@ -27,108 +27,247 @@ import answersUtilities from './answers-utilities';
+   * requests and an instance of the {@link AnswersCore} library.
+   *
+   * @param core - The Answers Core instance used for sending network requests
+   * @param stateManager - The manager of the state and how it can be manipulated
+   * @param httpManager - The manager for network requests and reponses through core
+   */
   constructor(
     private core: AnswersCore,
     private stateManager: StateManager,
     private httpManager: HttpManager,
   ) {}
 
+  /**
+   * Sets {@link QueryState.input} to the specified input.
+   *
+   * @param input - The new query input to update to
+   */
   setQuery(input: string): void {
     this.stateManager.dispatchEvent('query/setInput', input);
   }
 
+  /**
+   * Sets {@link QueryState.queryTrigger} to the specified trigger.
+   *
+   * @param trigger - The new query trigger to update to
+   */
   setQueryTrigger(trigger: QueryTrigger): void {
     this.stateManager.dispatchEvent('query/setTrigger', trigger);
   }
 
+  /**
+   * Sets {@link QueryState.querySource} to the specified source.
+   *
+   * @param source - The new query source to update to
+   */
   setQuerySource(source: QuerySource): void {
     this.stateManager.dispatchEvent('query/setSource', source);
   }
 
+  /**
+   * Sets {@link VerticalSearchState.verticalKey} to the specified key.
+   *
+   * @param verticalKey - The new vertical key to update to
+   */
   setVerticalKey(verticalKey: string): void {
     this.stateManager.dispatchEvent('vertical/setVerticalKey', verticalKey);
   }
 
+  /**
+   * Sets {@link VerticalSearchState.limit} to the specified limit.
+   *
+   * @param limit - The new vertical limit to update to
+   */
   setVerticalLimit(limit: number): void {
     this.stateManager.dispatchEvent('vertical/setLimit', limit);
   }
 
+  /**
+   * Sets {@link UniversalSearchState.limit} to the specified limit.
+   *
+   * @param limit - The new universal limit to update to
+   */
   setUniversalLimit(limit: UniversalLimit): void {
     this.stateManager.dispatchEvent('universal/setLimit', limit);
   }
 
+  /**
+   * Sets {@link VerticalSearchState.offset} to the specified offset.
+   *
+   * @param offset - The new vertical offset to update to
+   */
   setOffset(offset: number): void {
     this.stateManager.dispatchEvent('vertical/setOffset', offset);
   }
 
+  /**
+   * Sets {@link FiltersState."static"} to the specified filters.
+   *
+   * @param filters - The new static filters to update to
+   */
   setStaticFilters(filters: SelectableFilter[]): void {
     this.stateManager.dispatchEvent('filters/setStatic', filters);
   }
 
+  /**
+   * Sets {@link FiltersState.facets} to the specified facets.
+   *
+   * @param facets - The new facets to update to
+   */
   setFacets(facets: DisplayableFacet[]): void {
     this.stateManager.dispatchEvent('filters/setFacets', facets);
   }
 
+  /**
+   * Resets all {@link FiltersState.facets | facets} to be unselected.
```

Comment:
Thoughts on "Unselects all {@link FilterState.facets | facets}"?

Reply:
That works!